loadtest/control/browsercontroller/controller.go

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package browsercontroller
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/control"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/user"
+)
+
+// This is the URL of the LTBrowser API server ran from /browser
+const LT_BROWSER_API_URL = "http://localhost:5000"
+
+// BrowserController is a controller that manages browser sessions
+// by communicating with the browserLt API server.
+type BrowserController struct {
+	id              int
+	user            user.User
+	status          chan<- control.UserStatus
+	rate            float64
+	stopChan        chan struct{}
+	stoppedChan     chan struct{}
+	wg              *sync.WaitGroup
+	ltBrowserApiUrl string
+	httpClient      *http.Client
+	mu              sync.Mutex
+}
+
+// AddBrowserRequest represents the request body for adding a browser session
+type AddBrowserRequest struct {
+	// UserID is the username or email of the user to add a browser session
+	UserID   string `json:"userId" validate:"required"`
+	Password string `json:"password" validate:"required"`
+}
+
+// RemoveBrowserRequest represents the request body for removing a browser session
+type RemoveBrowserRequest struct {
+	// UserID is the username or email of the user to remove a browser session
+	UserID string `json:"userId" validate:"required"`
+}
+
+// BrowserAPIResponse represents the response from the browser API
+type BrowserAPIResponse struct {
+	Success bool   `json:"success"`
+	Message string `json:"message"`
+	Error   *struct {
+		Code    string `json:"code"`
+		Message string `json:"message"`
+	} `json:"error,omitempty"`
+}
+
+// New creates and initializes a new BrowserController with given parameters.
+func New(id int, user user.User, status chan<- control.UserStatus) (*BrowserController, error) {
+	if user == nil {
+		return nil, errors.New("user cannot be nil")
+	}
+	if status == nil {
+		return nil, errors.New("status channel cannot be nil")
+	}
+
+	return &BrowserController{
+		id:              id,
+		user:            user,
+		status:          status,
+		rate:            1.0,
```

Comment:
Commenting here as well in case my answer in the old thread is not visible: we don't really need to store the rate in this controller, we can simply implement SetRate as `SetRate(_ float64) error { return nil }`